### PR TITLE
refactor: reduce the number of external contract calls to DefisaverLo…

### PR DESCRIPTION
**Pull request by kjaredic**

### CODE REVIEW

File: contracts/actions/exchange/DFSBuy.sol [hunk 1]
```diff
@@ -50,16 +50,16 @@ contract DFSBuy is ActionBase, DFSExchangeCore {
         params.from = _parseParamAddr(params.from, _paramMapping[3], _subData, _returnValues);
         params.to = _parseParamAddr(params.to, _paramMapping[4], _subData, _returnValues);
 
-        uint256 exchangedAmount = _dfsBuy(params.exchangeData, params.from, params.to, RECIPE_FEE);
-
+        (uint256 exchangedAmount, bytes memory logData) = _dfsBuy(params.exchangeData, params.from, params.to, DIRECT_FEE);
```
Comment (yippee-ki-yay) [inline contracts/actions/exchange/DFSBuy.sol:53]: DIRECT _FEE used in recipe?

File: contracts/actions/exchange/DFSSell.sol [hunk 2]
```diff
@@ -51,16 +51,16 @@ contract DFSSell is ActionBase, DFSExchangeCore {
         params.from = _parseParamAddr(params.from, _paramMapping[3], _subData, _returnValues);
         params.to = _parseParamAddr(params.to, _paramMapping[4], _subData, _returnValues);
 
-        uint256 exchangedAmount = _dfsSell(params.exchangeData, params.from, params.to, RECIPE_FEE);
-
+        (uint256 exchangedAmount, bytes memory logData) = _dfsSell(params.exchangeData, params.from, params.to, DIRECT_FEE);
```
Comment (yippee-ki-yay) [inline contracts/actions/exchange/DFSSell.sol:54]: Should be RECIPE_FEE

File: contracts/actions/yearn/YearnWithdraw.sol [hunk 3]
```diff
@@ -39,15 +39,16 @@ contract YearnWithdraw is ActionBase {
...
         return (bytes32(amountReceived));
     }
 
     /// @inheritdoc ActionBase
     function executeActionDirect(bytes memory _callData) public payable override {
         Params memory inputData = parseInputs(_callData);
-
-        _yearnWithdraw(inputData);
+        (, bytes memory logData) = _yearnWithdraw(inputData);
+        logger.logActionDirectEvent("YearWithdraw", logData);
```
Comment (yippee-ki-yay) [inline contracts/actions/yearn/YearnWithdraw.sol:51]: Spelling mistake YearWithdraw?

**Review by yippee-ki-yay — CHANGES_REQUESTED** [review]
Fix the dfs sell/buy related error and a spelling mistake

File: contracts/actions/exchange/DFSBuy.sol (hunk 1 above)
Comment (kjaredic) [reply contracts/actions/exchange/DFSBuy.sol]: wild that this happened

File: contracts/actions/yearn/YearnWithdraw.sol (hunk 3 above)
Comment (kjaredic) [reply contracts/actions/yearn/YearnWithdraw.sol]: should've copy pasted

**Pull request merged**